test/Control/Effect/Spec.hs

```diff
@@ -1,122 +0,0 @@
-{-# LANGUAGE FlexibleContexts, FlexibleInstances, GeneralizedNewtypeDeriving, MultiParamTypeClasses, MultiWayIf, TemplateHaskell, TypeApplications, TypeOperators, UndecidableInstances #-}
-{-# OPTIONS_GHC -O2 -fplugin Test.Inspection.Plugin #-}
-module Control.Effect.Spec
-( spec
-) where
-
-import Control.Carrier
-import Control.Carrier.Error.Either
-import Control.Carrier.Fail.Either
-import Control.Carrier.Reader
-import Control.Carrier.State.Strict
-import Control.Effect.Sum
-import Prelude hiding (fail)
-import Test.Hspec
-import Test.Inspection as Inspection
-
-spec :: Spec
-spec = do
-  inference
-  reinterpretation
-  interposition
-  fusion
-
-inference :: Spec
-inference = describe "inference" $ do
-  it "can be wrapped for better type inference" $
-    run (runHasEnv (runEnv "i" ((++) <$> askEnv <*> askEnv))) `shouldBe` "ii"
```

Comment:
This is a pretty important technique and underserved here, so I moved it into the examples, above.